src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodToStateMachineRewriter.cs

```diff
@@ -253,10 +253,12 @@ protected BoundStatement GenerateHoistedLocalsCleanup(ImmutableArray<StateMachin
             // so that they can be collected by GC if needed
             foreach (var hoistedLocal in hoistedLocals)
             {
-                if (!hoistedLocal.Type.IsManagedType)
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                if (!hoistedLocal.Type.IsManagedType(ref useSiteDiagnostics))
                 {
                     continue;
                 }
+                F.Diagnostics.Add(hoistedLocal.Locations.FirstOrNone(), useSiteDiagnostics);
```

Comment:
do we have a test for this case?

Reply:
No, although looking this scenario, it seems like the only drawback of the missing references is that we zero out a field that might not need to be zeroed out (i.e. if it is of an unmanaged type). Therefore I am considering dropping the use site diagnostics in this scenario.

Reply:
I decided to keep the use site diagnostics here. It might end up being a breaking change, though.